File: LeetCodeTasks/13_roman-to-integer/solutions/java/solution.java
class Solution {
    public int romanToInt(String s) {
        int sum = 0;
        int prev = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            int curr = romanValue(s.charAt(i));
            if (curr < prev)
                sum -= curr;
            else
                sum += curr;
            prev = curr;
        }
        return sum;
    }

    private int romanValue(char r) {
        switch (r) {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
            default: return 0;
        }
    }
}

File: LeetCodeTasks/13_roman-to-integer/solutions/cpp/solution.cpp
#include <string>
#include <unordered_map>

class Solution {
public:
    int romanToInt(std::string s) {
        std::unordered_map<char, int> romanValues = {
            {'I', 1},
            {'V', 5},
            {'X', 10},
            {'L', 50},
            {'C', 100},
            {'D', 500},
            {'M', 1000}
        };
        
        int total = 0;
        int n = s.length();
        
        for (int i = 0; i < n; ++i) {
            int value = romanValues[s[i]];
            
            if (i + 1 < n && value < romanValues[s[i + 1]]) {
                total -= value;
            } else {
                total += value;
            }
        }
        
        return total;
    }
};
